test/puppetlabs/http/client/sync_ssl_test.clj

```diff
@@ -140,7 +140,8 @@
              (and (instance? SSLException cause#)
                   (or (re-find #"handshake_failure" message#)
                       (re-find #"internal_error" message#)))
-             (instance? ConnectionClosedException cause#))))))
+             (instance? ConnectionClosedException cause#))))
+     (catch ConnectionClosedException cce# true)))
```

Comment:
Not a huge fan of this, but the exception really doesn't contain any useful information to differentiate it from other reasons the connection might have been closed.

Reply:
huh. so it doesn't work to just use `java-unsupported-protocol-exception?` in the test, without your addition?

Reply:
No.  Apparently it isn't an `HttpClientException` when using the Clojure sync client, for some reason.  But maybe sometimes it is with older Java versions?  I'm kind of confused about that, and I couldn't suss out exactly where that split happens.

Reply:
So, I looked into this a bit and raising a HttpClientException is part of our API which Puppet Server depends on (to catch and re-raise the exception that the Puppet client is expecting. However, it only looks like this test fails like this when using the clojure client, not the Java client that we use in Puppet Server. It also doesn't seem like it's too big of deal if on very rare occasions we break that api contract because the outer layer of the Puppet http client blanket catches exceptions and wraps them in the correct exception.

tl;dr I don't think it would break the API in Puppet Server and if it did for some reason it probably would have a negligible impact.

Having said that, I'm going to look at this for the last half hour of my day and see if I can see why it only fails in the clojure wrapper of the client.

Reply:
So, I did discover that this is behavior is how the Java async client works, while the Java sync client catches any errors and wraps them in the HttpClientException and rethrows. The clojure sync client is a thin wrapper around the clojure async client which is then a thin wrapper around the java async client. Leaving only the java sync client to implement the desired behavior. Since that's what Puppet Server uses I think we can be confident that this isn't masking some issue in Puppet.

I think _at some point_ it'd be nice to make the other APIs wrap exceptions in HttpClientExceptions, but I don't think we need to block this on it.